amazonfreertossdk/src/main/java/software/amazon/freertos/amazonfreertossdk/AmazonFreeRTOSDevice.java

```diff
@@ -948,6 +953,7 @@ private void setMtu(int mtu) {
         if (mBleConnectionState == BleConnectionState.BLE_CONNECTED && mBluetoothGatt != null) {
             Log.i(TAG, "Setting mtu to: " + mtu);
             sendBleCommand(new BleCommand(REQUEST_MTU, mtu));
+            mBleConnectionStatusCallback.onBleConnectionStatusChanged(mBleConnectionState);
```

Comment:
why is the connection status change callback required here?

Reply:
BLE_CONNECTED is not useful to the application, since what is important is that the GATT service has been initialized. This is the reason to put the callback here.
It is probably better to add a new state, `BLE_INITIALIZED` or similar, which should notify the app that the BLE services have been initialized and are ready to be used. I would set that state in the `onMtuChanged` callback, and also call the application connection callback handler there.